### PR TITLE
memoDecode

### PR DESCRIPTION
**Pull request by technologiespro**
Added **memoDecode** method from history tx.

Test code:

```js
const BitShares =  require("./index.js");
BitShares.init("wss://bitshares.openledger.info/ws");
BitShares.subscribe('connected', start);

KEY = '5K...PrivateKey';
SENDER = 'sender_name';

async function start() {

    let bot = new BitShares(SENDER, KEY);
    bot.setMemoKey(KEY);
    let memo_text = await bot.memoDecode({
        "from": "BTS5...sender_address",
        "nonce": 192135435365768,
        "message": "b897254f..."
    });

    console.log(memo_text);

}
```

### CODE REVIEW

File: lib/bitshares.js [hunk 1]
```diff
@@ -198,6 +198,43 @@ export default class BitShares {
...
+        let memo_text;
+        let pub_key = await PublicKey.fromPublicKeyString(memos.from);
+        let prv_key = this.memoKey;
+        // console.log("\nPrivate key:", prv_key.toWif());
+        console.log("Public key :", prv_key.toPublicKey().toString(), "\n");
+
+        if (prv_key) {
+            try {
+                memo_text = prv_key
+                    ? await Aes.decrypt_with_checksum(
```
Comment (scientistnik) [inline lib/bitshares.js]: `if (prv_key)` and after that `prv_key ? ... : ...`? 
One condition excludes the other

File: lib/bitshares.js [hunk 2]
```diff
@@ -198,6 +198,43 @@ export default class BitShares {
...
+                console.log("transfer memo exception ...", e);
+                memo_text = false;
+            }
+
+        }
+
+
+        return {
+            message: memo_text
+        }
```
Comment (scientistnik) [inline lib/bitshares.js]: Why return an object?

Comment (technologiespro) [reply lib/bitshares.js]: oh yes, you are right, it is better to return the text and remove the superfluous condition

**Comment by technologiespro** [comment]
you are right, it is better to return the text and remove the superfluous condition

**Pull request merged**